### PR TITLE
#233: Support for live reload

### PR DESCRIPTION
**Pull request by AshwinKul28**
Via air package hot reload can be possible on different files such as .go, .tpl, .tmpl, .yaml.

Air package can be installed in many different ways mentioned here: https://github.com/air-verse/air

Best way if you're on go 1.22+
`go install github.com/air-verse/air@latest`

Or 

```
# binary will be $(go env GOPATH)/bin/air
curl -sSfL https://raw.githubusercontent.com/air-verse/air/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
```

### CODE REVIEW

**Comment by JyotinderSingh** [comment]
Please include instructions to use the hot reload feature in the README. Let's try to keep it as simple as possible to run the dev server (ideally a single command)

**Comment by AshwinKul28** [comment]
Added steps to install and run using the Air package. And yes once it's installed on the system, it's just a single-word command `air` in the desired directory. 

```
$ git clone https://github.com/dicedb/dice
$ cd dice
$ air
```

File: README.md [hunk 1]
```diff
@@ -40,6 +40,32 @@ $ cd dice
 $ go run main.go
 ```
 
+Enjoy a hussle free hot reload function using [Air package](https://github.com/air-verse/air)
```
Comment (JyotinderSingh) [inline README.md:43]: Add this section under subheading "Live Development Server"

Comment (JyotinderSingh) [inline README.md:43]:
```suggestion
DiceDB provides a hot-reloading development server, which allows you to instantly view your code changes in a live server. This functionality is supported by [Air](https://github.com/air-verse/air)
```

File: README.md [hunk 2]
```diff
@@ -40,6 +40,32 @@ $ cd dice
...
+
+To Install Air on your system you have following options.
+
+1. If you're on go 1.22+
+`go install github.com/air-verse/air@latest`
```
Comment (JyotinderSingh) [inline README.md:48]:
```suggestion
```sh
go install github.com/air-verse/air@latest
```
```

File: README.md [hunk 3]
```diff
@@ -40,6 +40,32 @@ $ cd dice
...
+
+
+
+2. Install using binary
```
Comment (JyotinderSingh) [inline README.md:52]:
```suggestion
2. Install the Air binary
```

File: README.md [hunk 4]
```diff
@@ -40,6 +40,32 @@ $ cd dice
...
+```sh
+# binary will be $(go env GOPATH)/bin/air
```
Comment (JyotinderSingh) [inline README.md:54]:
```suggestion
# binary will be installed at $(go env GOPATH)/bin/air
```

File: README.md [hunk 5]
```diff
@@ -40,6 +40,32 @@ $ cd dice
...
+curl -sSfL https://raw.githubusercontent.com/air-verse/air/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
+```
+
+Once `air` is installed you can verify using command `air -v`
```
Comment (JyotinderSingh) [inline README.md:58]:
```suggestion
Once `air` is installed you can verify the installation using the command `air -v`
```

File: README.md [hunk 6]
```diff
@@ -40,6 +40,32 @@ $ cd dice
...
+
+To run the DiceDB for local development:
```
Comment (JyotinderSingh) [inline README.md:60]:
```suggestion
To run the live DiceDB server for local development:
```

File: README.md [hunk 7]
```diff
@@ -40,6 +40,32 @@ $ cd dice
...
+
+```
```
Comment (JyotinderSingh) [inline README.md:62]:
```suggestion
```sh
```

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for this feature, have left some comments.

**Comment by AshwinKul28** [comment]
Requested changes are completed. Please check @JyotinderSingh thanks for the suggestions.

File: README.md [hunk 8]
```diff
@@ -40,6 +40,35 @@ $ cd dice
...
+### Live Development Server
+
+DiceDB provides a hot-reloading development environment, which allows you to instantly view your code changes in a live server. This functionality is supported by [Air](https://github.com/air-verse/air)
+
+To Install Air on your system you have following options.
+
+1. If you're on go 1.22+
+```sh
+go install github.com/air-verse/air@latest
+
```
Comment (JyotinderSingh) [inline README.md:52]: Add closing backticks

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]

**Review by JyotinderSingh — APPROVED** [review]
Thanks for adding this support, I'm sure it will improve the developer productivity on the project.

**Pull request merged**